### PR TITLE
Error checking for texture loading to fix green video on Linux (#647)

### LINKED ISSUE #647: Green screen instead of game intro
Когда начинаешь новую игру, вместо игровой заставки просто зелёный экран.

И такая же проблема в настройках. 
Снимок экрана_2020-07-12_17-50-52

**OS**
Slackware Current(x86_64)

**OpenXRay Version:**
730-july-preview

**OpenXRay Log:**
[openxray_nemial.log](https://github.com/OpenXRay/xray-16/files/4908882/openxray_nemial.log)

GOG сборка игры.

### PR DESCRIPTION
**Pull request by johncurley**
An uncaught error causes the game intro video to not load correctly due to a false positive on Linux and MacOS platforms. This fix allows the error stack to be popped before it reaches the error statement on the Linux platform.

### CODE REVIEW

File: src/Layers/xrRenderGL/glTexture.cpp [hunk 1]
```diff
@@ -121,21 +121,31 @@ GLuint CRender::texture_load(LPCSTR fRName, u32& ret_msize, GLenum& ret_desc)
...
         glm::tvec3<GLsizei> const tex_extent(texture.extent());
 
+        GLenum err;
         switch (texture.target())
         {
         case gli::TARGET_2D:
         case gli::TARGET_CUBE:
             CHK_GL(glTexStorage2D(target, static_cast<GLint>(texture.levels()), format.Internal,
                            tex_extent.x, tex_extent.y));
+            if ((err = glGetError()) != GL_NO_ERROR)
```
Comment (375gnu) [inline src/Layers/xrRenderGL/glTexture.cpp]: Remove CHK_GL to make this work (CHK_GL uses glGetError())

Comment (johncurley) [reply src/Layers/xrRenderGL/glTexture.cpp]: Isn't CHK_GL active only on Debug builds? This is happening on release builds.

File: src/Layers/xrRenderGL/glTexture.cpp [hunk 2]
```diff
@@ -121,21 +121,31 @@ GLuint CRender::texture_load(LPCSTR fRName, u32& ret_msize, GLenum& ret_desc)
...
+            }
             break;
         case gli::TARGET_3D:
+        case gli::TARGET_CUBE_ARRAY:
             CHK_GL(glTexStorage3D(target, static_cast<GLint>(texture.levels()), format.Internal,
                            tex_extent.x, tex_extent.y, tex_extent.z));
+            if ((err = glGetError()) != GL_NO_ERROR)
+            {
+                Msg("Error: 0x%x: Invalid 3D texture: '%s'", err, fname);
+            }
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderGL/glTexture.cpp]: Here's what I suggest:
```suggestion
            glTexStorage3D(target, static_cast<GLint>(texture.levels()), format.Internal,
                           tex_extent.x, tex_extent.y, tex_extent.z);
            const auto err = glGetError();
            if (err != GL_NO_ERROR)
            {
                VERIFY(err == GL_NO_ERROR, "Error when loading a texture");
                Msg("! OpenGL: 0x%x: Invalid 3D texture: '%s'", err, fname);
            }
```

File: src/Layers/xrRenderGL/glTexture.cpp (hunk 1 above)
Comment (375gnu) [reply src/Layers/xrRenderGL/glTexture.cpp]: I believe it's better to have a consistent behavior in all builds.

File: src/Layers/xrRenderGL/glTexture.cpp [hunk 3]
```diff
@@ -6,6 +6,9 @@
 
 #include <gli/gli.hpp>
 
+//#include "xrCore/xrDebug_macros.h"
+
+
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderGL/glTexture.cpp]:
```suggestion
```

File: src/Layers/xrRenderGL/glTexture.cpp [hunk 4]
```diff
@@ -121,21 +124,35 @@ GLuint CRender::texture_load(LPCSTR fRName, u32& ret_msize, GLenum& ret_desc)
...
+            if (err != GL_NO_ERROR)
+            {
+                VERIFY(err == GL_NO_ERROR);
+                Msg("! OpenGL: 0x%x: Invalid 2D texture: '%s'", err, fname);
+            }
             break;
         case gli::TARGET_3D:
-            CHK_GL(glTexStorage3D(target, static_cast<GLint>(texture.levels()), format.Internal,
-                           tex_extent.x, tex_extent.y, tex_extent.z));
+        case gli::TARGET_CUBE_ARRAY:
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderGL/glTexture.cpp]:
```suggestion
```

File: src/Layers/xrRenderGL/glTexture.cpp [hunk 5]
```diff
@@ -160,35 +177,61 @@ GLuint CRender::texture_load(LPCSTR fRName, u32& ret_msize, GLenum& ret_desc)
...
+                            {
+                                VERIFY(err == GL_NO_ERROR);
+                                Msg("! OpenGL: 0x%x: Invalid 2D subtexture: '%s'", err, fname);
+                            }
+
                         }
                         break;
                     }
                     case gli::TARGET_3D:
+                    case gli::TARGET_CUBE_ARRAY:
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderGL/glTexture.cpp]:
```suggestion
```

**Review by Xottab-DUTY — APPROVED** [review]

**Comment by Xottab-DUTY** [comment]
Thanks, @johncurley!

It the commit [9030849](https://github.com/OpenXRay/xray-16/pull/1225/commits/9030849ff60aded1aea919f9d45da861e709bf40) I have removed newly added switch cases `gli::TARGET_CUBE_ARRAY` so that this pull requests only fixes a bug and doesn't add anything new.
Can you add these switch cases in a new PR?

**Comment by johncurley** [comment]
No worries. Would you like me to create a new PR with that in it?

**Comment by Xottab-DUTY** [comment]
@johncurley, yes :)

**Pull request merged**